compiler/src/dmd/escape.d

```diff
@@ -36,6 +36,27 @@ import dmd.tokens;
 import dmd.visitor;
 import dmd.arraytypes;
 
+/// Groups global state for escape checking together
+package(dmd) struct EscapeState
+{
+    // Maps `sequenceNumber` of a `VarDeclaration` to an object that contains the
+    // reason it failed to infer `scope`
+    // https://issues.dlang.org/show_bug.cgi?id=23295
+    private __gshared RootObject[int] scopeInferFailure;
+
+    /// Called by `initDMD` to initialize the escape state.
+    static void initialize()
+    {
+        scopeInferFailure = null;
+    }
+
+    /// Called by `deinitializeDMD` to deinitialize the escape state.
+    static void deinitialize()
```

Comment:
Is there really any point in having both these functions? Can't we have just the `deinitalize` one and skip calling `initialize`? If not, can't we give it a more generic name like `reset` ?

Reply:
I made it consistent with the other aggregates in `initDMD` and `deinitializeDMD`.

Reply:
It's now a single `reset` function